Assets/Mirror/Tests/Editor/MessageTests.cs

```diff
@@ -24,6 +24,45 @@ public void ErrorMessageTest()
             Assert.That(writer.ToArray()[0], Is.EqualTo(123));
         }
 
+        [Test]
+        public void SpawnMessageTest()
+        {
+            // try setting value with constructor
+            SpawnMessage message = new SpawnMessage
+            {
+                netId = 42,
+                isLocalPlayer = true,
+                isOwner = true,
+                sceneId = 42,
+                assetId = Guid.NewGuid(),
+                position = UnityEngine.Vector3.one,
+                rotation = UnityEngine.Quaternion.identity,
+                scale = UnityEngine.Vector3.one,
+                payload = new ArraySegment<byte>(new byte[] { 0x01, 0x02 })
+            };
+
+            // serialize
+            NetworkWriter writer = new NetworkWriter();
+            message.Serialize(writer);
+            byte[] writerData = writer.ToArray();
+
+            // deserialize the same data - do we get the same result?
+            SpawnMessage fresh = new SpawnMessage();
+            fresh.Deserialize(new NetworkReader(writerData));
+            Assert.That(fresh.netId, Is.EqualTo(message.netId));
+            Assert.That(fresh.isLocalPlayer, Is.EqualTo(message.isLocalPlayer));
+            Assert.That(fresh.isOwner, Is.EqualTo(message.isOwner));
+            Assert.That(fresh.sceneId, Is.EqualTo(message.sceneId));
+            Assert.That(fresh.assetId, Is.EqualTo(message.assetId));
```

Comment:
SpawnMessageTest (0.008s)
---
Expected: edf5bc1b-bd03-40ae-9431-3b7830fe31bd
  But was:  00000000-0000-0000-0000-000000000000
---
at Mirror.Tests.MessageTests.SpawnMessageTest () [0x00115] in D:\Unity\Projects\Mirror2019-3\Assets\Mirror\Tests\Editor\MessageTests.cs:56

Reply:
```cs
public struct SpawnMessage : IMessageBase
{
    public uint netId;
    public bool isLocalPlayer;
    public bool isOwner;
    public ulong sceneId;
    public Guid assetId;
    public Vector3 position;
    public Quaternion rotation;
    public Vector3 scale;
    // the serialized component data
    // -> ArraySegment to avoid unnecessary allocations
    public ArraySegment<byte> payload;
```